Dockerfile.windows

```diff
@@ -152,15 +153,14 @@
 # -----------------------------------------------------------------------------------------
 
 
+# Extremely important. If using this Dockerfile in process isolated containers,
+# the kernel of the host must match the container image, and hence would fail
+# between Windows Server 2016 (aka RS1) and Windows Server 2019 (aka RS5).
+ARG WINDOWS_VERSION=${WINDOWS_VERSION:-1809}
+
 # The number of build steps below are explicitly minimised to improve performance.
 
-# Extremely important - do not change the following line to reference a "specific" image, 
```

Comment:
Hmmm. Aren't you effectively doing exactly this with hard-coding `ARG` to 1809 above???

Reply:
It can be overridden, but we need some default so that `docker build` works without _requiring_ a `--build-arg`

Reply:
I can confirm on a Windows Server 2016 I can build it with
```
docker build -t nativebuildimage --build-arg WINDOWS_VERSION=1607 -f Dockerfile.windows .
```
and it pulls the image with the 1607 tag.

Reply:
But
```
ARG WINDOWS_VERSION=1809
FROM mcr.microsoft.com/windows/servercore:${WINDOWS_VERSION}
``` 
would also work as a default and can be overwritten with `--build-arg`. This is how I use it in other Dockerfiles.